tests/test_custom_extension.py

```diff
@@ -74,3 +76,26 @@ def test_generated_extension(tmpfolder):
     with chdir("my_project"):
         # Testing a project generated by the custom extension
         run_common_tasks(tests=False)
+
+
+def test_new_project_does_not_fail_pre_commit(cwd, pre_commit, putup):
+    # Given pyscaffold is installed,
+    # when we call putup with extensions and pre-commit
+    name = "my_project"
+    run(f"{putup} --pre-commit --dsproject -p my_package {name}")
```

Comment:
Should we test using a `--namespace my.pkg` here so we ensure the worst case scenario is working?

Reply:
Good point! I fixed this in the last commit.